### ISSUE TITLE
bug: `allOf` when an element is a `oneOf` union generates an invalid zod schema

### ISSUE THREAD

**Issue by mnahkies** [mnahkies/openapi-code-generator#282]
Given a schema like:

```yaml
components:
  schemas:
    Example:
      allOf:
        - oneOf:
            - type: object
              properties:
                foo:
                  type: string
            - type: object
              properties:
                bar:
                  type: string
        - type: object
          properties:
            id:
              type: string
```

Which produces the type:
```typescript
export type t_Example = (
  | {
      foo?: string
    }
  | {
      bar?: string
    }
) & {
  id?: string
}
```

We get an invalid `zod` schema, as `.merge` isn't supported by `ZodUnion`

**Comment by mnahkies** [comment]
Note: this is also an issue with the `joi` schema builder.

**Comment by mnahkies** [comment]
Fixed for `zod` by #283, still an issue for `joi` (https://github.com/hapijs/joi/issues/3057) but going to leave this as a known issue / todo in the code for now.
